setup.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2016-2018, The University of Texas at Austin
+# & University of California, Merced.
+#
+# All Rights reserved.
+# See file COPYRIGHT for details.
+#
+# This file is part of the hIPPYlib library. For more information and source code
+# availability see https://hippylib.github.io.
+#
+# hIPPYlib is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License (as published by the Free
+# Software Foundation) version 2.0 dated June 1991.
+
+
+from setuptools import setup, find_packages
+from os import path
+from io import open
+
+
+try:
+    from dolfin import __version__ as dolfin_version
+except ImportError:
+    raise
+else:
+    if (int(dolfin_version[0:4]) < 2016):
+        raise Exception(
+            "hippylib requires FEniCS installation not older than 2016.1.0")
+
+here = path.abspath(path.dirname(__file__))
+with open(path.join(here, 'README.md'), encoding='utf-8') as f:
+    long_description = f.read()
+
+REQUIREMENTS = [
+    "mpi4py",
+    "numpy",
+    "matplotlib",
+    "scipy",
+    "sympy==1.1.1",
```

Comment:
I would not mention `sympy==1.1.1` since it is a requirement for FEniCS.
The issue is that the `conda-forge` receipt for FEniCS 2017.2 pulls the latest version of sympy, which breaks FEniCS form compiler.

Reply:
@joshuawchen ,

I kind of agree that  `matplotlib` and `jupyter` are not real requirements, but not having them really kills the user experience. Is there any keyword for *recommended* dependencies that will be installed by default unless the user specifies otherwise?

Reply:
@uvilla @joshuawchen I do agree with not requiring `jupyter` (I personally don't use it), however I think `matplotlib` should be kept as required, as there is a part of the library (`nb` IIRC) that relies on it, so it has to be a dependence.

I will look if there is a way to make "optional" dependencies or something.